hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/volume/HddsVolume.java

```diff
@@ -470,4 +461,22 @@ private void closeDbStore() {
     LOG.info("SchemaV3 db is stopped at {} for volume {}", containerDBPath,
         getStorageID());
   }
+
+  public boolean compactDb() {
+    File dbDir = getDbParentDir();
+    File dbFile = new File(dbDir, CONTAINER_DB_NAME);
+    if (dbFile.exists() && dbFile.canRead()) {
```

Comment:
There is no need to check exists and canRead, which is covered in check() already.  If it's checked here, then you should mark the volume as failure if check failed. This boolean result is not used any where.